birdhouse/components/weaver/config/magpie/weaver_hooks.py

```diff
@@ -30,7 +30,21 @@
 
     from magpie.adapter import HookContext
 
-LOGGER = get_logger("birdhouse-weaver-hooks")
+LOGGER = get_logger("magpie.weaver-hooks")
+
+# NOTE:
+#  This value should correspond to the directory defined for public data on WPS outputs.
+#  The output hierarchy would be:
+#
+#       ${WEAVER_WPS_OUTPUTS_DIR}/
+#           users/
+#               <user-id>/
+#                   <job-id>/
+#           ${PUBLIC_USER_CONTEXT}/
+#               <job-id>/
+#
+# NOTE: This value should align with Cowbird's setting 'PUBLIC_WORKSPACE_WPS_OUTPUTS_SUBDIR'.
```

Comment:
Instead of adding this note. Why don't we actually use the `PUBLIC_WORKSPACE_WPS_OUTPUTS_SUBDIR` variable to set this constant?

If we make this a `.template` file then we can just insert the correct value to ensure that it aligns.

Reply:
Good idea. 

I think it would be wise also to mask away the nesting logic of `PUBLIC_WORKSPACE_WPS_OUTPUTS_SUBPATH` here.
Technically, the original path was `public` only, but Cowbird adjusted recently to be `public/wps_outputs` in prevision that other "public" stuff might be used. 
I think we should "help" the user such that if `public` or `${PUBLIC_WORKSPACE_WPS_OUTPUTS_SUBPATH}` is passed, both will redirect to `${PUBLIC_WORKSPACE_WPS_OUTPUTS_SUBPATH}`. This way, they can just request "make my output public" and not worry about instance-specific configuration.

Sounds good?

Reply:
Yes that's a great idea!